### ISSUE TITLE
typeMapper returned ENUM type missing name

### ISSUE THREAD

**Issue by jarno-steeman** [mickhansen/graphql-sequelize#20]
Every GraphQL object is required to have a unique name iirc, but the returned GraphQLEnumType from the typeMapper doesnt provide one. 

The first GraphQLEnumType created with name `undefined` by the typeMapper will not throw an error, but the second time it'll give you an error saying that a GraphQL object with name `undefined` already exists.

**Comment by jarno-steeman** [comment]
The typeMapper should probably receive the whole attribute instead of just the type, to get access to the Model's name and the attribute's name.

**Comment by mickhansen** [comment]
Would not make it a typeMapper anymore though ;)
But we'll probably have to remove ENUM from the typeMapper and do a special case in attributeFields.

**Comment by Aweary** [comment]
Does it matter what the name is for the `GraphQLEnumType`? If we just need to enforce unique names we can generate unique random strings for the name.

**Comment by jarno-steeman** [comment]
It will make the returned schema valid again since name's are just required to be unique, but introspection of your schema saying Object x has a field of type "Di57gpl" (random string) isn't very nice.

**Comment by Aweary** [comment]
Well, looking nice in introspection is a secondary concern to functionality. So I think what it should do is generate a random string for the schema name, note that it does so in the docs, and if you really need to have specific declared names you can exclude the `ENUM` and define it yourself.

**Comment by brad-decker** [comment]
In order to maintain the function of typeMapper only being a typeMapper, we could check to see if the returned type from typeMapper is a ENUM inside of attributeFields and add the name after the fact using key and model name. So something like [modelname][Key]EnumType - userStatusEnumType ?

**Comment by jarno-steeman** [comment]
The unique name checking takes place inside the object constructor so that wouldnt fix the error.

**Comment by brad-decker** [comment]
The unique name error is thrown from the schema object, not from the GraphQLEnumType/Or Object Types. GraphQL doesn't know its non-unique until you try to build the schema.

**Comment by brad-decker** [comment]
```
Error: Schema must contain unique named types but contains multiple types named "undefined".
    at invariant ([...]/node_modules/graphql/jsutils/invariant.js:20:11)
    at typeMapReducer ([...]/node_modules/graphql/type/schema.js:138:41)
```

**Comment by jarno-steeman** [comment]
Ahh, yes I see. Then i'd say run like that. It makes more sense to implement the name logic inside the attributeFields function.

**Comment by brad-decker** [comment]
@Aweary + @mickhansen thoughts on this solution? Requires an additional type check inside of attributeFields which is the only negative.

**Comment by Aweary** [comment]
:+1:

**Comment by mickhansen** [comment]
@brad-decker Sounds like a good solution for now.

**Comment by brad-decker** [comment]
@mickhansen i threw the fix for this into my open Pull request. added tests for coverage

**Comment by mickhansen** [comment]
Published in `v0.9.0`

**Comment by jlaramie** [comment]
@mickhansen this should be reopened as the fix does not cover nesting via `DataTypes.ARRAY(DataTypes.ENUM(ENUM_ARRAY))`
